packages/core/src/core/communication.ts

```diff
@@ -205,3 +205,50 @@ export async function createRepository(
       throw error;
     });
 }
+
+const UserProfile = t.exact(
```

Comment:
We already have that logic in SM-UI.
I don't think this should be in the Core (debate ongoing on slack).
Either
- we put it in the Core and make SM-UI use it.
- we duplicate the code from SM-UI to the init